File: day01/py/d1.py
text_file = open("../input.txt", "r")
lines = [v.strip("\n") for v in text_file.readlines()]

sumlist=[]
acc=0

for i,line in enumerate(lines):
    if line != "" or i==len(lines): #you're not at the end of the elf, so keep summing
        acc += int(line)
    else: #you're at the end of the elf, so append the sum and reset it
        sumlist.append(acc)
        acc=0

print("Part 1:",max(sumlist))
sumlist.sort()
print("Part 2:",sum(sumlist[-3:]))

File: day01/R/d1.R
rm(list=ls())
input=readLines("../input.txt")

bundles = c()
acc=0
for(line in input){ #iterate over each line in input
  if(line != ""){ #if line is not missing then you're still on the same elf...
    acc=acc+as.numeric(line) #...so add value to the running sum
  }else{ #otherwise you've hit the end of the elf... 
    bundles=c(bundles,acc) #...so append to vector of sums per elf
    acc=0 #and reset running sum
  }
}

print(paste("Part 1:",max(bundles))) #max
print(paste("Part 2:",sum(tail(sort(bundles),3)))) #get sum of top 3